### PR TITLE
bugfix: Don't read file for InteractiveSemanticdbdb if it doesn't exist

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would assume that any file that we want to calculate InteractiveSemanticdb for exists on the filesystem, which is not true. For example renamed files if renamed from terminal will stick in the editor and cause exceptions to be throw. Now, we also try to read it from buffers and from filesystem as a fallback if it exists on the filesystem.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala [hunk 1]
```diff
@@ -87,18 +85,25 @@ final class InteractiveSemanticdbs(
...
-            Try(compile(path, text)) match {
-              case Success(doc) if doc != null =>
-                if (!source.isDependencySource(workspace))
-                  semanticdbIndexer().onChange(source, doc)
-                doc
-              case _ => null
-            }
-          } else
-            existingDoc
+          buffers.get(source).orElse {
```
Comment (filipwiech) [inline metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala:88]: Sorry to bother you, but out of curiosity, what is the difference between the `unsavedContents` (which by the way is still used above in the `shouldTryCalculateInteractiveSemanticdb`, not sure if that's intended now) and `buffers.get(source)`? :thinking:

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala]: Och, actually `unsavedContents` is unused and it should be used :scream: We use it for to warm up scalafix at the start :thinking: 

I changed it as a first option with fallback to buffers then to filesystem.

File: metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala [hunk 2]
```diff
@@ -87,18 +85,25 @@ final class InteractiveSemanticdbs(
...
-              case Success(doc) if doc != null =>
-                if (!source.isDependencySource(workspace))
-                  semanticdbIndexer().onChange(source, doc)
-                doc
-              case _ => null
-            }
-          } else
-            existingDoc
+          unsavedContents.orElse(buffers.get(source).orElse {
+            if (source.exists) Some(source.readText)
```
Comment (jkciesluk) [inline metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala:89]: Should we care about non-standard  charset?

**Review by jkciesluk — APPROVED** [review]
lgtm

File: metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala (hunk 2 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala]: I haven't seen this being an issue, but you're right. Fixed it.

**Pull request merged**